### ISSUE TITLE
[feature request] put tensor shape and type back in printout

### ISSUE THREAD

**Issue by meder411** [pytorch/pytorch#7115]
I'm sure there's some reason as to why the tensor shape (dimensions) and type are no longer included (as of 0.4) when `print x` is called on some tensor `x`, but why? It can be extremely useful, especially when viewing the contents of large tensors. Could this information be re-added in a future release?

**Comment by zou3519** [comment]
cc @li-roy

**Comment by meder411** [comment]
I know it's only one guy's opinion, but I actually really preferred the old printout all around. I see you guys moving to be more like NumPy and TensorFlow with this change, but I actually find it harder to infer the axis I'm looking at based on brackets a la

```
tensor([[ 2.,  6.,  5.,  4.,  3.,  0.,  5.,  4.,  7.,  3.,  4.,  7.,
          6.,  3.,  7.,  4.,  6.,  0.,  3.,  4.],
        [ 3.,  6.,  0.,  2.,  6.,  0.,  4.,  6.,  0.,  1.,  0.,  1.,
          3.,  0.,  2.,  3.,  2.,  4.,  0.,  2.]], device='cuda:0')
```

than I did when they were explicitly described in the printing as they were <0.4. Just my few cents.

**Comment by gchanan** [comment]
@meder411 I suggested some possible improvements over in https://github.com/pytorch/pytorch/issues/6912, you may want to weigh in on the discussion over there.
